jscomp/runtime/caml_obj.ml

```diff
@@ -223,6 +248,16 @@ and aux_length_b_short (a : Obj.t) (b : Obj.t) i short_length =
     let res = caml_compare (Obj.field a i) (Obj.field b i) in
     if res <> 0 then res
     else aux_length_b_short a b (i+1) short_length
+and aux_obj_compare (a: Obj.t) keys_a (b: Obj.t) keys_b i min_len default_res =
+  if i = min_len then default_res
```

Comment:
if the length is not the same, should we exit early here? 
Note it does not to have to follow the semantics of caml_compare for block  which is constrained by existing ocaml specification.

Oh, one thing which reminds of me that `caml_compare` semantics will be changed if we change record from array to dict is that the iteration order needs to be the same as declaration order

Reply:
Found a way to avoid Object.keys entirely, and only rely on for..in, the code structure will change.